header-rewrite-filter/header_processor.cc

```diff
@@ -428,24 +468,25 @@ namespace HeaderRewriteFilter {
         return absl::InvalidArgumentError("invalid syntax for dynamic function -- function too short");
     }
 
-    // make sure dynamic function is wrapped in %[]
     if (function_expression.substr(0, 2) != Utility::DYNAMIC_FUNCTION_DELIMITER.substr(0, 2) && function_expression.substr(function_expression.size()-1, 1) != Utility::DYNAMIC_FUNCTION_DELIMITER.substr(2, 1)) {
-        // TODO: treat it as a static value if it's not wrapped in %[]
-        return absl::InvalidArgumentError("dynamic function must be wrapped with %[]");
-    }
-    function_type_ = getFunctionType(function_expression.substr(2, function_expression.size() - Utility::DYNAMIC_FUNCTION_DELIMITER.size()));
-    if (function_type_ == Utility::FunctionType::InvalidFunctionType) {
-        return absl::InvalidArgumentError("invalid function type for dynamic value");
-    }
-    if (function_type_ == Utility::FunctionType::Urlp && !is_request_) {
-        return absl::InvalidArgumentError("cannot get url path parameter on response side");
-    }
-    const std::tuple<absl::Status, std::string> get_function_argument_result = getFunctionArgument(function_expression.substr(2, function_expression.size() - Utility::DYNAMIC_FUNCTION_DELIMITER.size()));
-    const absl::Status status = std::get<0>(get_function_argument_result);
-    if (status != absl::OkStatus()) {
-        return status;
+        function_type_ = Utility::FunctionType::Static;
+        function_argument_ = std::string(function_expression);
+        return absl::OkStatus();
+    } else {
```

Comment:
Nit: because the above `if` returns at the end, you could remove the `else` and reduce indentation